program-runtime/src/invoke_context.rs

```diff
@@ -146,10 +146,21 @@ impl BpfAllocator {
 
 pub struct SyscallContext {
     pub allocator: BpfAllocator,
-    pub orig_account_lengths: Vec<usize>,
+    pub accounts_metadata: Vec<SerializedAccountMetadata>,
     pub trace_log: Vec<[u64; 12]>,
 }
 
+#[derive(Debug, Clone)]
+pub struct SerializedAccountMetadata {
+    pub original_data_len: usize,
+    pub is_writable: bool,
```

Comment:
Why store `is_writable` and `is_executable`?
They won't change after the feature [`B9cdB55u4jQsDNsdTK525yE9dmSc5Ga7YBaBrDFvEhM9`](https://github.com/solana-labs/solana/issues/26988) is enabled.

Reply:
Because those are populated when execute() calls serialize_parameters(). The latter iterates over all the accounts, borrows them, builds the memory regions and the account metadata. The account metadata includes those two flags.

If we don't store those two flags there, then in execute() we need to do another loop over all the accounts, borrow them again, fetch those two flags and store them in another allocation. Seemed unnecessarily wasteful to me.

Reply:
Why? The flags are only used in the error handler for write access violation, right? We could just borrow that one account there and get its flags. I am just worried about spreading redundant data over multiple structs.

Reply:
Yeah I had tried that, but with the closure I had a borrowing error and so gave up too soon. I've removed the closure and managed to make it work now, so I removed those two fields from SerializedAccountMetadata